diff --git a/Libraries/Components/View/ViewAccessibility.d.ts b/Libraries/Components/View/ViewAccessibility.d.ts
index c4f462739efb0041e05fe304f6e7cedaf3ae441b..c4ea02315ec613b87e4014569c3da20e35cec8d5 100644
--- a/Libraries/Components/View/ViewAccessibility.d.ts
+++ b/Libraries/Components/View/ViewAccessibility.d.ts
@@ -357,9 +357,11 @@ export type Role =
   | 'application'
   | 'article'
   | 'banner'
+  | 'blockquote'
   | 'button'
   | 'cell'
   | 'checkbox'
+  | 'code'
   | 'columnheader'
   | 'combobox'
   | 'complementary'
diff --git a/types_generated/Libraries/Components/View/ViewAccessibility.d.ts b/types_generated/Libraries/Components/View/ViewAccessibility.d.ts
index 1c29b54b5fafe133403d6f50ce94cf7551ae569d..92b6b962b65a2138f24b101c091c94c53f973f3b 100644
--- a/types_generated/Libraries/Components/View/ViewAccessibility.d.ts
+++ b/types_generated/Libraries/Components/View/ViewAccessibility.d.ts
@@ -12,7 +12,7 @@
 
 import type { NativeSyntheticEvent } from "../../Types/CoreEventTypes";
 export type AccessibilityRole = "none" | "button" | "dropdownlist" | "togglebutton" | "link" | "search" | "image" | "keyboardkey" | "text" | "adjustable" | "imagebutton" | "header" | "summary" | "alert" | "checkbox" | "combobox" | "menu" | "menubar" | "menuitem" | "progressbar" | "radio" | "radiogroup" | "scrollbar" | "spinbutton" | "switch" | "tab" | "tabbar" | "tablist" | "timer" | "list" | "toolbar" | "grid" | "pager" | "scrollview" | "horizontalscrollview" | "viewgroup" | "webview" | "drawerlayout" | "slidingdrawer" | "iconmenu" | string;
-export type Role = "alert" | "alertdialog" | "application" | "article" | "banner" | "button" | "cell" | "checkbox" | "columnheader" | "combobox" | "complementary" | "contentinfo" | "definition" | "dialog" | "directory" | "document" | "feed" | "figure" | "form" | "grid" | "group" | "heading" | "img" | "link" | "list" | "listitem" | "log" | "main" | "marquee" | "math" | "menu" | "menubar" | "menuitem" | "meter" | "navigation" | "none" | "note" | "option" | "presentation" | "progressbar" | "radio" | "radiogroup" | "region" | "row" | "rowgroup" | "rowheader" | "scrollbar" | "searchbox" | "separator" | "slider" | "spinbutton" | "status" | "summary" | "switch" | "tab" | "table" | "tablist" | "tabpanel" | "term" | "timer" | "toolbar" | "tooltip" | "tree" | "treegrid" | "treeitem";
+export type Role = "alert" | "alertdialog" | "application" | "article" | "banner" | "blockquote" | "button" | "cell" | "checkbox" | "code" | "columnheader" | "combobox" | "complementary" | "contentinfo" | "definition" | "dialog" | "directory" | "document" | "feed" | "figure" | "form" | "grid" | "group" | "heading" | "img" | "link" | "list" | "listitem" | "log" | "main" | "marquee" | "math" | "menu" | "menubar" | "menuitem" | "meter" | "navigation" | "none" | "note" | "option" | "presentation" | "progressbar" | "radio" | "radiogroup" | "region" | "row" | "rowgroup" | "rowheader" | "scrollbar" | "searchbox" | "separator" | "slider" | "spinbutton" | "status" | "summary" | "switch" | "tab" | "table" | "tablist" | "tabpanel" | "term" | "timer" | "toolbar" | "tooltip" | "tree" | "treegrid" | "treeitem";
 export type AccessibilityActionName =
 /**
  * Generated when a screen reader user double taps the component.
